apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/FirestoreDb.cs

```diff
@@ -201,5 +201,21 @@ string ExtractPath(DocumentReference documentReference)
                 return documentReference.Path;
             }
         }
+
+        // TODO: Naming? Java has getCollections
+
+        /// <summary>
+        /// Asynchronously retrieves the root collection IDs from the server.
+        /// </summary>
+        /// <param name="cancellationToken">A cancellation token for the operation.</param>
+        /// <returns>All the root collection IDs.</returns>
+        public Task<IList<CollectionReference>> ListRootCollectionsAsync(CancellationToken cancellationToken = default) =>
+            ListCollectionsAsync(null, cancellationToken);
+
+        internal async Task<IList<CollectionReference>> ListCollectionsAsync(DocumentReference parent, CancellationToken cancellationToken) =>
```

Comment:
Is exposing this as a `Task<IList...>` the right thing to do? Rather than as an async collection of some sort?

Is the size of the returned collection always/usually small enough that fetching the whole thing is reasonable?

Reply:
The return type is as per the specification I've got - but looking at the Java code, that only seems to create a single request. I'll clarify this with the team.

We could potentially use "Get" as the prefix for now with this return type, and add "List" later on that returns a paged enumerable.

Reply:
Yes, I think using a `Get` prefix for this is a good idea.